src/vtpty/SshSession.cpp

```diff
@@ -0,0 +1,1302 @@
+class SshPtySlave final: public PtySlave
+{
+  public:
+    void close() override {}
+    [[nodiscard]] bool isClosed() const noexcept override { return false; }
+    bool configure() noexcept override { return true; }
+    bool login() override { return true; }
+    int write(std::string_view) noexcept override { return 0; }
+};
+// }}}
+
+#if defined(_WIN32)
+// Why is it, that Windows API is so much more complicated than POSIX? Extrawurst!
+using socket_handle = crispy::native_handle<SOCKET, INVALID_SOCKET>;
+#else
+using socket_handle = crispy::native_handle<int, -1>;
+#endif
+
+std::string SshHostConfig::toString() const
+{
+    auto result = ""s;
+    auto const add = [&](std::string_view text) {
+        if (!text.empty())
+        {
+            if (!result.empty())
+                result += ", ";
+            result += text;
+        }
+    };
+    if (!username.empty())
+        result += fmt::format("{}@", username);
+    if (!hostname.empty())
+    {
+        if (port)
+        {
+            if (hostname.find(':') != std::string::npos)
+                result += fmt::format("[{}]:{}"sv, hostname, port);
+            else
+                result += fmt::format("{}:{}"sv, hostname, port);
+        }
+        else
+            result += hostname;
+    }
+    else if (port)
+        result += "*:" + std::to_string(port);
+
+    if (!privateKeyFile.empty())
+        add(fmt::format("private key: {}", privateKeyFile.string()));
+
+    if (!publicKeyFile.empty())
+        add(fmt::format("public key: {}", privateKeyFile.string()));
+
+    if (!knownHostsFile.empty())
+        add(fmt::format("known hosts: {}", knownHostsFile.string()));
+
+    add(fmt::format("ForwardAgent: {}", forwardAgent ? "Yes" : "No"));
+
+    return result;
+}
+
+std::string SshHostConfig::toConfigString(std::string const& host) const
+{
+    std::string result;
+    auto const prefix = host.empty() ? ""sv : "  "sv;
+
+    if (!host.empty())
+        result += fmt::format("Host {}\n", host);
+
+    if (!hostname.empty())
+        result += fmt::format("{}HostName {}\n", prefix, hostname);
+    if (port != 22)
+        result += fmt::format("{}Port {}\n", prefix, port);
+    if (!username.empty())
+        result += fmt::format("{}User {}\n", prefix, username);
+    if (!privateKeyFile.empty())
+        result += fmt::format("{}IdentityFile {}\n", prefix, privateKeyFile.string());
+    if (!knownHostsFile.empty())
+        result += fmt::format("{}KnownHostsFile {}\n", prefix, knownHostsFile.string());
+    result += fmt::format("{}ForwardAgent {}\n", prefix, forwardAgent);
+    result += fmt::format("\n");
+    return result;
+}
+
+SshHostConfigMap loadSshConfig(std::filesystem::path const& configPath)
+{
+    std::ifstream file(configPath);
+    SshHostConfigMap configs;
+    SshHostConfig defaultConfig;
+    std::string line;
+    std::string currentHost;
+
+    if (!file)
+        throw std::runtime_error("Could not open the SSH config file.");
```

Comment:
It means that we can not spawn ssh session, and mandate user to have `.ssh/config` file i am not sure that we want to do that, we should be able to spawn ssh session without .ssh/config file

Reply:
I will fix it in a minute / moment. Thanks